projects/vdk-core/src/vdk/internal/builtin_plugins/config/log_config.py

```diff
@@ -65,6 +67,30 @@ def _parse_log_level_module(log_level_module):
         )
 
 
+def configure_initial_logging_before_anything():
+    """
+    Configure logging at the start of the app.
+    At this point we do not know what user has configured so we use some sensible approach
+    All logs would be vdk only logs. User code will not have been executed yet so we will use log_level_vdk
+    to control the level.
+    We default to WARN since in most cases users would not care about internal vdk logs.
+
+    Logging lifecycle:
+    1. This function is executed at the entry point of the app and configures default logging format and level
+    2. After vdk_initialize hook is executed and CLI command starts then `vdk --verbosity` option is taken (if set)
+        and click logging configuration takes over
+    3. When initialize_job hook (applicable for vdk run only) is executed then below configure_loggers function
+        is run which adds more context to the logs and initializes syslog handler (if configured to do so)
+    """
+    log_level = "WARNING"
+    if os.environ.get("LOG_LEVEL_VDK", None):
```

Comment:
Why not using the imported `LOG_LEVEL_VDK` constant, imported at the top of the file?

Reply:
No reason, not to use it.